assembly_uploader/ena_queries.py

```diff
@@ -86,36 +53,134 @@ def parse_accession(accession):
 
 
 class EnaQuery:
-    def __init__(self, accession, username=None, password=None):
-        self.url = ENA_API_URL
+    def __init__(self, accession, private=False):
+        self.private_url = "https://www.ebi.ac.uk/ena/submit/report/"
+        self.public_url = "https://www.ebi.ac.uk/ena/portal/api/search"
         self.accession = accession
         self.acc_type = parse_accession(accession)
-        username = username or os.getenv("ENA_WEBIN")
-        password = password or os.getenv("ENA_WEBIN_PASSWORD")
+        username = os.getenv("ENA_WEBIN")
+        password = os.getenv("ENA_WEBIN_PASSWORD")
+        if username is None or password is None:
+            logging.error("ENA_WEBIN and ENA_WEBIN_PASSWORD are not set")
         if username and password:
             self.auth = (username, password)
         else:
             self.auth = None
-        self.data = build_data(self.accession, self.acc_type)
-
-    def post_request(self):
-        if self.auth:
-            response = requests.post(
-                self.url,
-                data=self.data,
-                auth=self.auth,
-                **get_default_connection_headers(),
+        self.private = private
+
+    def post_request(self, data):
+        response = requests.post(
+            self.public_url, data=data, **get_default_connection_headers()
+        )
+        return response
+
+    def get_request(self, url):
+        response = requests.get(url, auth=self.auth)
+        return response
+
+    def check_api_error(self, response):
+        try:
+            data = json.loads(response.text)[0]
+            return data
+        except NoDataException:
```

Comment:
I don't think `NoDataException` will ever be raised, unless I am missing it somewhere?

Reply:
Hmm, I still think this case will never be raised?